sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java

```diff
@@ -195,43 +227,46 @@ public SubscriptionPath subscriptionPath() {
     return subscriptionPath;
   }
 
-  private List<SubscriptionPath> listSubscriptions(ProjectPath projectPath, TopicPath topicPath)
-      throws IOException {
-    return pubsub.listSubscriptions(projectPath, topicPath).stream()
-        .filter((path) -> !path.equals(subscriptionPath))
-        .collect(ImmutableList.toImmutableList());
+  private Iterable<String> listSubscriptions(TopicPath topicPath) {
+    assert topicAdmin != null;
+    return topicAdmin.listTopicSubscriptions(topicPath.getPath()).iterateAll();
   }
 
   /** Publish messages to {@link #topicPath()}. */
-  public void publish(List<PubsubMessage> messages) throws IOException {
-    List<PubsubClient.OutgoingMessage> outgoingMessages =
-        messages.stream().map(this::toOutgoingMessage).collect(toList());
-    pubsub.publish(eventsTopicPath, outgoingMessages);
-  }
+  public void publish(List<PubsubMessage> messages) {
+    assert eventsTopicPath != null;
```

Comment:
Assert isn't guaranteed to run. Would something like `org.apache.beam.sdk.util.Preconditions.checkArgumentNotNull` be more appropriate?

Reply:
Good point, done